homeassistant/components/ecobee/climate.py

```diff
@@ -601,18 +584,12 @@ def set_preset_mode(self, preset_mode: str) -> None:
             else:
                 _LOGGER.warning("Received unknown preset mode: %s", preset_mode)
 
-        else:
```

Comment:
With the introduction of validation, we're guaranteed that the provided `preset_mode` is present in `self.preset_modes` making this block irrelevant. The validation also means that none of the presets removed above could be passed in, making all of this code even more unnecessary.